pkg/apis/batch/v1alpha1/job.go

```diff
@@ -107,6 +107,34 @@ type JobSpec struct {
 	// +kubebuilder:validation:Minimum=1
 	// +optional
 	MinSuccess *int32 `json:"minSuccess,omitempty" protobuf:"varint,12,opt,name=minSuccess"`
+
+	// NetworkTopologies defines the NetworkTopologies config, this field works in conjunction with network topology feature and hyperNode CRD.
+	// +optional
+	NetworkTopologies *NetworkTopologiesSpec `json:"networkTopologies,omitempty" protobuf:"bytes,13,opt,name=networkTopologies"`
+}
+
+type NetworkTopologyMode string
+
+const (
+	// HardNetworkTopologyMode represents a strict network topology constraint that jobs must adhere to.
+	HardNetworkTopologyMode NetworkTopologyMode = "hard"
+
+	// SoftNetworkTopologyMode represents a flexible network topology constraint that allows jobs
+	// to cross network boundaries under certain conditions.
+	SoftNetworkTopologyMode NetworkTopologyMode = "soft"
+)
+
+type NetworkTopologiesSpec struct {
+	// Mode specifies the mode of the network topology constrain.
+	// +kubebuilder:validation:Enum=hard;soft
+	// +kubebuilder:default=hard
+	// +optional
+	Mode NetworkTopologyMode `json:"mode,omitempty" protobuf:"bytes,1,opt,name=mode"`
+
+	// HighestTierAllowed specifies the highest tier that a job allowed to cross when scheduling.
+	// +kubebuilder:default=1
+	// +optional
+	HighestTierAllowed int `json:"highestTierAllowed,omitempty" protobuf:"bytes,2,opt,name=highestTierAllowed"`
```

Comment:
```suggestion
	HighestTierAllowed *int32 `json:"highestTierAllowed,omitempty" protobuf:"bytes,2,opt,name=highestTierAllowed"`
```
When CR update, Controller will get new value.

Reply:
You mean it will be override when update?

Reply:
> You mean it will be override when update?

Yes. It's possible

Reply:
Sorry I'm a little confused, I'm not sure you point: ), The difference between value and pointer type is we can exactly know that whether a field is not set or set a zero vlaue, in this case if HighestTierAllowed  is not present, it means that this field is not set by user when the type is *int32.

Reply:
Already set to *int.